### PR TITLE
Refresh Scope on restart.

### PR DESCRIPTION
**Pull request by OlgaMaciaszek**

### CODE REVIEW

File: spring-cloud-context/src/main/java/org/springframework/cloud/context/refresh/RefreshScopeLifecycle.java [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+	public RefreshScopeLifecycle(ContextRefresher contextRefresher) {
+		this.contextRefresher = contextRefresher;
+	}
+
+	@Override
+	public void start() {
+		synchronized (lifecycleMonitor) {
+			if (!isRunning()) {
+				if (LOG.isDebugEnabled()) {
+					LOG.debug("Refreshing context on restart.");
```
Comment (spencergibb) [inline spring-cloud-context/src/main/java/org/springframework/cloud/context/refresh/RefreshScopeLifecycle.java:46]: I'd say this is important enough for info rather than debug

**Review by spencergibb — APPROVED** [review]

**Review by spencergibb — APPROVED** [review]

**Comment by OlgaMaciaszek** [comment]
@Buzzardo could you please take a look at the doc changes?

File: docs/modules/ROOT/pages/spring-cloud-commons/application-context-services.adoc [hunk 2]
```diff
@@ -217,6 +217,12 @@ The configuration property must be present in order to update the value after a
 a value in your application you might want to switch your logic to rely on its absence instead.  Another option would be to rely
 on the value changing rather than not being present in the application's configuration.
 
+[refresh-scope-on-restart]
+=== Refresh Scope on Restart
+
+In order to allow seamlessly refreshing beans on restart, which is especially useful for applications running with JVM Checkpoint Restore (for example with https://github.com/CRaC[Project CRaC]), we now instantiate a `RefreshScopeLifecycle` bean that will trigger Context Refresh on restart, resulting in rebinding configuration properties and refreshing any `@RefreshScope`-annotated beans. This behaviour can be disabled by setting the value of `spring.cloud.refresh.on-restart.enabled` to `false`.
```
Comment (Buzzardo) [inline docs/modules/ROOT/pages/spring-cloud-commons/application-context-services.adoc:223]: That long and complex sentence is hard to parse in one pass, so please change it to: 

Seamlessly refreshing beans on restart is especially useful for applications that run with JVM Checkpoint Restore (such as https://github.com/CRaC[Project CRaC]). To allow this ability, we now instantiate a `RefreshScopeLifecycle` bean that triggers Context Refresh on restart, resulting in rebinding configuration properties and refreshing any beans annotated with `@RefreshScope`. You can disable this behavior by setting `spring.cloud.refresh.on-restart.enabled` to `false`.

Comment (OlgaMaciaszek) [reply docs/modules/ROOT/pages/spring-cloud-commons/application-context-services.adoc:223]: Thanks @Buzzardo - will change.
